Automattic/wp-calypso
# Stylelint: properly scope all disable comments

### Proposed Changes
Work towards #67467. For every unscoped `stylelint-disable`, scope it to a specific rule. Additionally, replace uses of `stylelint-disable` with `stylelint-disable-line`. Unfortunately, there is [no built-in way to enforce this](https://github.com/stylelint/stylelint/issues/2292) going forward, but `--fix` [does not work correctly](https://github.com/stylelint/stylelint/issues/2643) if rules are unscoped. (E.g. if you write `/* stylelint-disable-line */`, the **entire file** cannot be autofixed. Autofix only works for the rest of the file if the line is scoped to a rule.)

It will be very important moving forward to make sure that autofix works, and therefore forbid the use of disable lines which break it. Maybe we can write a basic shell script or plugin that can do that without too much trouble, but it's out of scope for this PR.

The specific motivation for this PR is that we cannot autofix the 88 impacted files if they use unscoped disable rules.

### Testing Instructions
There's no easy way to divide this up and review it other than just looking at the files and making sure I didn't make egregious syntax errors. It's also not crucial for every rule to be exact -- as we move forward with fixing style issues, any problems with these changes will be discovered and fixed.

You can also test by:
- `yarn lint:css` on trunk vs on this branch and verify there are no regressions.
- Search across the code base for unscoped rules and make sure I didn't miss any. (e.g. "/* stylelint-disable-line */")
